index.d.ts

```diff
@@ -9,9 +9,12 @@ export interface Connection extends mysql.Connection {
     execute<T extends mysql.RowDataPacket[][] | mysql.RowDataPacket[] | mysql.OkPacket | mysql.OkPacket[]>(options: mysql.QueryOptions, callback?: (err: mysql.QueryError | null, result: T, fields?: mysql.FieldPacket[]) => any): mysql.Query;
     execute<T extends mysql.RowDataPacket[][] | mysql.RowDataPacket[] | mysql.OkPacket | mysql.OkPacket[]>(options: mysql.QueryOptions, values: any | any[] | { [param: string]: any }, callback?: (err: mysql.QueryError | null, result: T, fields: mysql.FieldPacket[]) => any): mysql.Query;
     ping(callback?: (err: mysql.QueryError | null) => any): void;
+    promise(promiseImpl?: PromiseConstructor): PromiseConnection;
 }
 
-export interface PoolConnection extends mysql.PoolConnection, Connection {}
+export interface PoolConnection extends mysql.PoolConnection, Connection {
+    promise(promiseImpl?: PromiseConstructor): PromisePoolConnection;
```

Comment:
I'm not at my desk right now so I can't test this, but are you sure this declaration properly overrides the `Connection.promise` method? I've had some weird results in similar situations in the past.